src/router_core/connections.c

```diff
@@ -305,12 +307,14 @@ int qdr_connection_process(qdr_connection_t *conn)
 
     int event_count = 0;
 
+    sys_mutex_lock(conn->work_lock);
```

Comment:
I did not think very hard if this is the correct lock to use. It seemed conveniently placed, so I just extended what it protects by a few lines up. Actually, the `closed` flag seems to be tightly bound to the list of links with work, so it may make good sense this way, in fact.

Reply:
@kgiusti approved this and the issue has manifested on GHA in controlled environment; therefore, issue is real and fix should be acceptable!